test/YesSql.Tests/SqlServerTests.cs

```diff
@@ -442,5 +442,96 @@ public async Task ShouldIndexPropertyKeysWithBits()
                 session.Save(property);
             }
         }
+
+        [Fact]
+        public async Task AbleToCreateGuidAsUniqueIdentifier()
```

Comment:
What if the test is in CoreTests instead? Such that it's tested on all database and not just SQL Server?
I assume that's why it was mapped to a string in the first place. Or the solution should take that into account and behave different on each database, maybe by having type mappings defined in the dialect.

Reply:
Yes I can move it up. I just don't have instance of every database to run the test on. But, I can move it to CoreTests. Would you want me to send it a second pull request and you can run a full test on your side?

Reply:
Just move it up in this PR, the CI will do it for you. I don't have any database on my local system ;) I rely on the PR CI.

Reply:
@sebastienros as you guessed the test failed on one of the database providers :( I was able to run the test on MySQL and SQL Server database and it worked. I am guessing it is not working on other providers. Not sure if there is a way to tell which provider it is failing for. Is there a way to fix this issue so that Guid is treated as Guid not as a string?

Reply:
It's in the build log on the CI, where you took the message from. In this case Sqlite. If it had passed it would have then tried with PostgresQL. 
It might require to move the dictionary you changed to the Dialect base class, and let each dialect define custom mappings for the types. That was expected.
You can try it if you want, it shouldn't be complex since I assume all the code paths that need the dictionary already have access to the dialect.